File: GhostLight/GhostLight C#/GhostLight/GhostLight/MenueSystem/TextWithNewLines.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Drawing;
using CustomWindower.CoreEngine;

/**
 * Is Identical to Text in every way
 * Except when you put '\n' into a string It makes the rest of the String a new Line
 * @author Michael Letter
 */
namespace MenueSystem{
    public class TextWithNewLines : Text{
        public override void paint(Camera drawLocation) {
            String[] totalText = null;
            string oldText = "";
            if(visible && drawLocation != null && text != null && !text.Equals("")){
                if(!oldText.Equals(base.text) || totalText == null){
                    totalText = base.text.Split(new Char[] { '\n' });
                    oldText = text;
                }
	            float startHeight = base.textPosition.Y;
	            float lineWidth = 3f;
                updateTransform();

                for(int loop = 0; loop < totalText.Length; loop++){
                    textPosition.Y = textPosition.Y + (lineWidth * loop);
                    if (targetFont != null && targetFont.isLoaded()){
                        targetFont.paintText(drawLocation, totalText[loop], textPosition, textColor, textTransform);
                    }
                }
                textPosition.Y = startHeight;
            }
	    }
    }
}

File: Corrupted/CorruptedOOAPI/src/MenueSystem/TextWithNewLines.java
package MenueSystem;

import Engine.Text;

/**
 * Is Identical to Text in every way
 * Except when you put '\n' into a string It makes the rest of the String a new Line
 * @author Michael Letter
 */
//Gahh!!
public class TextWithNewLines extends Text{

	public void draw(){
	    String totalText = super.getText();
	    float startHeight = super.center.getY();
	    float lineWidth = 3f;
	    int lastNewLine = 0;
	    while(!super.getText().equals("")){
	    	int newLineIndex = totalText.indexOf('\n', lastNewLine + 1);
	    	if(newLineIndex >= 0){
	    		super.setText(totalText.substring(lastNewLine, newLineIndex));
	    		super.draw();
	    		super.center.setY(super.center.getY() - lineWidth);
	    		lastNewLine = newLineIndex;
	    	}
	    	else if(lastNewLine > 0 && lastNewLine != totalText.length()){
	    		super.setText(totalText.substring(lastNewLine, totalText.length()));
	    		super.draw();
	    		break;
	    	}
	    	else{
	    		 super.draw();
	    		 break;
	    	}
	    }
	    super.center.setY(startHeight);
	    super.setText(totalText);
	}
}
